src/main/java/org/mockito/internal/creation/bytebuddy/InlineByteBuddyMockMaker.java

```diff
@@ -202,16 +207,25 @@ public InlineByteBuddyMockMaker() {
         }
         bytecodeGenerator =
                 new TypeCachingBytecodeGenerator(
-                        new InlineBytecodeGenerator(INSTRUMENTATION, mocks, mockedStatics), true);
+                        new InlineBytecodeGenerator(
+                                INSTRUMENTATION, mocks, mockedStatics, mockConstruction::get),
+                        true);
     }
 
     @Override
     public <T> T createMock(MockCreationSettings<T> settings, MockHandler handler) {
         Class<? extends T> type = createMockType(settings);
 
-        Instantiator instantiator = Plugins.getInstantiatorProvider().getInstantiator(settings);
         try {
-            T instance = instantiator.newInstance(type);
+            T instance;
+            try {
+                // We attempt to use the "native" mock maker first that avoids Objenesis and Unsafe
```

Comment:
In which case can this fail and do we have to fallback to `Objenesis`? E.g. ideally if we go for this pattern and we can avoid `Objenesis`, I would like to remove it as a dependency as a whole.

Reply:
We could do that in a second phase.